pipelinedp4j/tests/com/google/privacy/differentialprivacy/pipelinedp4j/spark/SparkTableTest.kt

```
@@ -0,0 +1,178 @@
+        val dataset = spark.createDataset(listOf(), Encoders.tuple(Encoders.STRING(), Encoders.INT()))
+        val sparkTable = SparkTable(dataset, Encoders.STRING(), Encoders.INT())
+        val result = sparkTable.keysEncoder
+
+        assertThat(result).isInstanceOf(SparkEncoder::class.java)
+        assertThat(result.encoder).isEqualTo(Encoders.STRING())
+    }
+
+    @Test
+    fun valuesEncoder_returnsCorrectEncoder() {
+        val dataset = spark.createDataset(listOf(), Encoders.tuple(Encoders.STRING(), Encoders.INT()))
+        val sparkTable = SparkTable(dataset, Encoders.STRING(), Encoders.INT())
+        val result = sparkTable.valuesEncoder
+
+        assertThat(result).isInstanceOf(SparkEncoder::class.java)
+        assertThat(result.encoder).isEqualTo(Encoders.INT())
+    }
+
+    @Test
+    fun map_appliesMapFn() {
+        val dataset = spark.createDataset(listOf(Tuple2(1, 10)), Encoders.tuple(Encoders.INT(), Encoders.INT()))
+        val sparkTable = SparkTable(dataset, Encoders.INT(), Encoders.INT())
+        val mapFn: (Int, Int) -> String = { k, v -> "${k}_$v" }
+        val result = sparkTable.map("Test", sparkEncoderFactory.strings(), mapFn)
+        assertThat(result.data.collectAsList()).containsExactly("1_10")
+    }
+
+    @Test
+    fun groupAndCombineValues_appliesCombiner() {
+        val dataset = spark.createDataset(listOf(Tuple2("positive", 1),
+            Tuple2("positive", 10), Tuple2("negative", -1),
+            Tuple2("negative", -10)
+        ), Encoders.tuple(Encoders.STRING(), Encoders.INT()))
+        val sparkTable = SparkTable(dataset, Encoders.STRING(), Encoders.INT())
+        val combineFn: (Int, Int) -> Int = { v1, v2 -> v1 + v2 }
+        val result = sparkTable.groupAndCombineValues("Test", combineFn)
+        assertThat(result.data.collectAsList()).containsExactly(Tuple2("positive", 11), Tuple2("negative", -11))
+    }
+
+    @Test
+    fun groupByKey_groupsValues() {
+
```

should we throw an exception for now so that we don't forget to implement it once we support it? (if we are going to support it in this PR)